tests/transforms/test_adjoint_metric_tensor.py

```diff
@@ -0,0 +1,704 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the adjoint metric tensor function.
+"""
+import pytest
+from pennylane import numpy as np
+import pennylane as qml
+
+from pennylane.transforms.adjoint_metric_tensor import _apply_any_operation
+
+
+class TestApplyAnyOperation:
```

Comment:
Do trainable templates also need to be tested here?

Reply:
Yeah, maybe that's a good call :)

Reply:
I added a test for the metric tensor itself that uses a template. For `_apply_operations` it does not make sense because it would need to be decomposed first, which happens earlier on in the pipeline :)